packages/core/services/fit-bounds.ts

```diff
@@ -0,0 +1,71 @@
+import { MapsAPILoader } from './maps-api-loader/maps-api-loader';
+import { LatLngLiteral, LatLngBounds, LatLng, LatLngBoundsLiteral } from './google-maps-types';
+import { Observable } from 'rxjs/Observable';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { from } from 'rxjs/observable/from';
+import {
+  skipWhile,
+  flatMap,
+  mergeMap,
+  debounce,
+  map,
+  distinctUntilChanged
+} from 'rxjs/operators';
+import { timer } from 'rxjs/observable/timer';
+import { Subject } from 'rxjs/Subject';
+
+declare var google: any;
+
+export interface FitBoundsDetails {
+  latLng: LatLng|LatLngLiteral;
+}
+
+export abstract class FitBoundsAccessor {
+  abstract getFitBoundsDetails$(): Observable<FitBoundsDetails>;
+}
+
+@Injectable()
+export class FitBoundsService {
+  readonly bounds$: Observable<LatLngBounds>;
+  protected readonly _boundsChangeDebounceTime$: BehaviorSubject<number> = new BehaviorSubject<number>(200);
+  protected readonly _includeInBounds$: BehaviorSubject<Map<string, LatLng | LatLngLiteral>> = new BehaviorSubject<Map<string, LatLng | LatLngLiteral>>(new Map<string, LatLng | LatLngLiteral>());
+  protected _emitPaused: boolean = false;
+
+  constructor(loader: MapsAPILoader) {
+    this.bounds$ = from(loader.load()).pipe(
+      flatMap(() => this._includeInBounds$),
+      skipWhile(() => this._emitPaused),
+      // debounce(() => this._boundsChangeDebounceTime$),
```

Comment:
`_boundsChangeDebounceTime$` is not needed, right?

Reply:
no only fitbounds, thank you!

Reply:
@boogie77 it's not finished yet